dpgen2/exploration/deviation/deviation_spin.py

```
@@ -0,0 +1,127 @@
+            self.MAX_DEVI_MF,
+            self.MIN_DEVI_MF,
+            self.AVG_DEVI_MF,
+        )
+        # check the length of model deviations
+        frames = {}
+        for name in model_devi_names:
+            if len(self._data[name]) > 0:
+                assert len(self._data[name]) == self.ntraj, (
+                    f"Error: the number of model deviation {name} "
+                    + f"({len(self._data[name])}) and trajectory files ({self.ntraj}) "
+                    + f"are not equal."
```

Ruff

f-string without any placeholders

Remove extraneous `f` prefix

(F541)